mrmustard/physics/wires.py

```diff
@@ -1,31 +1,169 @@
-# Copyright 2023 Xanadu Quantum Technologies Inc.
+"""Module containing wire classes for quantum and classical channels."""
 
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
+from __future__ import annotations
+from functools import cached_property
+from enum import Enum, auto
+from typing import Any, Iterable, Iterator
+from dataclasses import dataclass, field, replace
+from random import randint
+from IPython.display import display
+from mrmustard import widgets
 
-#     http://www.apache.org/licenses/LICENSE-2.0
 
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
+__all__ = ["Wires"]
 
-"""``Wires`` class for supporting tensor network functionalities."""
+"""
+This module provides wire functionality for applications in MrMustard.
+It defines the core classes for representing quantum and classical wires, and their
+relationships in quantum optical circuits.
+"""
 
-from __future__ import annotations
-from functools import cached_property
-import numpy as np
 
-from IPython.display import display
+class LegibleEnum(Enum):
+    """Enum class that provides a more legible string representation."""
 
-from mrmustard import widgets
+    def __str__(self) -> str:
+        return self.name
 
-__all__ = ["Wires"]
+    def __repr__(self) -> str:
+        return self.name
```

Comment:
I'm not sure about the need for this in the repr.  The default str is probably not too bad either as it's just the enum with the instance name.

Reply:
I just dislike the mess that precedes the name